src/Util/HTTPSignature.php

```diff
@@ -537,14 +538,12 @@ public static function fetchRaw(string $request, int $uid = 0, array $opts = [Ht
 
 		if (!empty($uid)) {
 			$owner = User::getOwnerDataById($uid);
-			if (!$owner) {
-				return;
-			}
 		} else {
 			$owner = User::getSystemAccount();
-			if (!$owner) {
-				return;
-			}
+		}
+
+		if (!$owner) {
+			throw new Exception('Could not find owner for uid ' . $uid);
```

Comment:
This probably should log a critical error about the system account not being set up when `$uid` is empty.

Reply:
I've checked every calling places. All places are calling it inside a try-catch block and are logging an exception like this:

```php
try {
	$curlResult = HTTPSignature::fetchRaw($request, $uid);
} catch (\Exception $exception) {
	DI::logger()->notice('Error fetching url', ['url' => $request, 'exception' => $exception]);
}
```

As one can see the `$exception` is always logged too, so throwing this new exception should be enough context.